src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsde.agora.server.business
+
+import org.broadinstitute.dsde.agora.server.AgoraConfig
+import org.broadinstitute.dsde.agora.server.dataaccess.AgoraDao
+import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+
+/**
+ * Created by dshiga on 5/13/15.
+ */
+object AgoraBusiness extends AgoraDao {
+
+  def agoraUrl(entity: AgoraEntity): String = {
+    AgoraConfig.methodsUrl + entity.namespace.get + "/" + entity.name.get + "/" + entity.snapshotId.get
+  }
+
+  def addUrl(entity: AgoraEntity): AgoraEntity = {
+    entity.copy(url = Option(agoraUrl(entity)))
+  }
+
+  override def insert(agoraEntity: AgoraEntity): AgoraEntity = {
+    AgoraDao.createAgoraDao.insert(agoraEntity).copy(url = Option(agoraUrl(agoraEntity)))
```

Comment:
I'm not against it. As they would say "Mutability is an optimization". I'm fine leaving it as is but we should note that this might be a target for optimization should performance become an issue.